tests/backup/backup_helper.go

```diff
@@ -1426,3 +1435,200 @@ func DeleteBackupAndWait(backupName string, ctx context.Context) error {
 	_, err := task.DoRetryWithTimeout(backupDeletionSuccessCheck, backupDeleteTimeout, backupDeleteRetryTime)
 	return err
 }
+
+// GetPxBackupVersion returns the version of Px Backup like 2.4.0-e85b680
+func GetPxBackupVersion() (string, error) {
+	ctx, err := backup.GetAdminCtxFromSecret()
+	if err != nil {
+		return "", err
+	}
+	versionResponse, err := Inst().Backup.GetPxBackupVersion(ctx, &api.VersionGetRequest{})
+	if err != nil {
+		return "", err
+	}
+	version := versionResponse.GetVersion()
+	return fmt.Sprintf("%s.%s.%s-%s", version.GetMajor(), version.GetMinor(), version.GetPatch(), version.GetGitCommit()), nil
+}
+
+// GetPxBackupVersionSemVer returns the version of Px Backup in semver format like 2.4.0
+func GetPxBackupVersionSemVer() (string, error) {
+	ctx, err := backup.GetAdminCtxFromSecret()
+	if err != nil {
+		return "", err
+	}
+	versionResponse, err := Inst().Backup.GetPxBackupVersion(ctx, &api.VersionGetRequest{})
+	if err != nil {
+		return "", err
+	}
+	version := versionResponse.GetVersion()
+	return fmt.Sprintf("%s.%s.%s", version.GetMajor(), version.GetMinor(), version.GetPatch()), nil
+}
+
+// GetPxBackupBuildDate returns the Px Backup build date
+func GetPxBackupBuildDate() (string, error) {
+	ctx, err := backup.GetAdminCtxFromSecret()
+	if err != nil {
+		return "", err
+	}
+	versionResponse, err := Inst().Backup.GetPxBackupVersion(ctx, &api.VersionGetRequest{})
+	if err != nil {
+		return "", err
+	}
+	version := versionResponse.GetVersion()
+	return version.GetBuildDate(), nil
+}
+
+// UpgradePxBackup will perform the upgrade tasks for Px Backup to the version passed as string
+// Eg: versionToUpgrade := "2.4.0"
+func UpgradePxBackup(versionToUpgrade string) error {
+	var cmd string
+
+	// Compare and validate the upgrade path
+	currentBackupVersionString, err := GetPxBackupVersionSemVer()
+	if err != nil {
+		return err
+	}
+	currentBackupVersion, err := version.NewSemver(currentBackupVersionString)
```

Comment:
I think we can directly return Semver with GetPxBackupVersionSemVer()

Reply:
But in `GetPxBackupVersionSemVer()` the return type is string, to do `GreaterThanOrEqual` the expected type is `Version`